zeebe/msgpack-value/src/main/java/io/camunda/zeebe/msgpack/property/BaseProperty.java

```diff
@@ -98,16 +103,48 @@ public void write(final MsgPackWriter writer) {
     valueToWrite.write(writer);
   }
 
-  public void writeJSON(final StringBuilder sb) {
+  public void writeJSON(final StringBuilder sb, final boolean maskSanitized) {
     key.writeJSON(sb);
     sb.append(":");
     if (hasValue()) {
-      resolveValue().writeJSON(sb);
+      if (maskSanitized && isSanitized) {
+        sb.append("\"***\"");
+      } else {
+        resolveValue().writeJSON(sb);
+      }
     } else {
       sb.append("\"NO VALID WRITEABLE VALUE\"");
     }
   }
 
+  public boolean isSanitized() {
+    return isSanitized;
+  }
+
+  /**
+   * If set to true, will be replaced by '***' when writing to console or logs, i.e. when called via
+   * {@link #toString()}, or {@link #writeJSON(StringBuilder, boolean)} with the second param set to
+   * true.
+   *
+   * <p>If false, disabled sanitization of this value in the cases above.
+   *
+   * <p>The chaining portion is a bit hacky, but it allows you to declare fields as:
+   *
+   * <pre>{@code
+   * private final IntegerProperty myProperty = new IntegerProperty("myProperty").sanitized();
+   * }</pre>
+   *
+   * The proper way for chaining would be to add a type param to the class that would represent the
+   * current type, but that would create quite a lot of refactoring, and this is likely good enough
+   * in most cases. Feel free to improve on this.
```

Comment:
😆 This reminded me on the meme and I was looking for a counter, "how many people tried to refactored this"